### PR TITLE
Feature/image saving callback

### PR DESCRIPTION
**Pull request by benjijamorris**
## What does this PR do?
- move image saving from each head to a ImageSave callback
- remove `save-input` argument from im2im heads
- move all example im2im configs to using an image saving callback

# PLUGIN DEVS
This should preserve format except for saved out image filenames (which I don't think we're relying on anymore?) Feel free to chime in if you have more questions/tests I should run. 

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?
- [x] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: configs/experiment/im2im/segmentation.yaml [hunk 1]
```diff
@@ -34,3 +34,19 @@ data:
...
+callbacks:
+  # prediction
+  # saving:
+  #   _target_: cyto_dl.callbacks.image_saver.ImageSaver
+  #   save_dir: ${paths.output_dir}
+  #   save_every_n_epochs: ${model.save_images_every_n_epochs}
+  #   stages: ["predict"]
+  #   save_input: False
+  # training
+  saving:
```
Comment (ritvikvasan) [inline configs/experiment/im2im/segmentation.yaml:47]: what is the saving key here? Shouldnt callbacks just be a list of callbacks?

File: cyto_dl/nn/head/gan_head.py [hunk 2]
```diff
@@ -80,13 +77,18 @@ def run_head(
...
-            "y_hat_out": y_hat_out,
-            "y_out": y_out,
+            "pred": self._postprocess(y_hat, img_type="prediction", n_postprocess=n_postprocess),
+            "target": self._postprocess(
+                batch[self.head_name], img_type="input", n_postprocess=n_postprocess
+            )
+            if stage != "predict"
+            else None,
+            "input": self._postprocess(
+                batch[self.x_key], img_type="input", n_postprocess=n_postprocess
```
Comment (ritvikvasan) [inline cyto_dl/nn/head/gan_head.py:90]: what is "img_type" here?

**Review by ritvikvasan — APPROVED** [review]
Looks good to me!

File: configs/experiment/im2im/segmentation.yaml (hunk 1 above)
Comment (benjijamorris) [reply configs/experiment/im2im/segmentation.yaml:47]: they all have a key : https://github.com/AllenCellModeling/cyto-dl/blob/323ff267fa061b3eace107ba44248677e878902c/configs/callbacks/default.yaml#L8

File: cyto_dl/nn/head/gan_head.py (hunk 2 above)
Comment (benjijamorris) [reply cyto_dl/nn/head/gan_head.py:90]: in the config I broke down postprocesssing by whether it's applied to the input/target/prediction

**Review by ritvikvasan — APPROVED** [review]
LGTM!

**Pull request merged**